File: leetsync/3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.go
func minimumOperations(nums []int) int {
	n := len(nums)
	freq := make(map[int]int)
	for i := n - 1; i >= 0; i-- {
		freq[nums[i]]++
		if freq[nums[i]] > 1 {
			return (i + 3) / 3
		}
	}
	return 0
}

File: leetsync/3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.java
class Solution {
    public int minimumOperations(int[] nums) {
        int n = nums.length;
        int[] map = new int[101];
        for (int i = n - 1; i >= 0; i--) {
            if (++map[nums[i]] > 1) {
                return (i + 3) / 3;
            }
        }
        return 0;
    }
}